content/courses/connecting-to-offchain-data/oracles.md

```diff
@@ -12,255 +12,252 @@ description: Access real-world data inside a Solana program.
+staked amount of the oracle network, there may still be sufficient incentive for
+collusion among oracles.
+
+As a developer, it is your responsibility to understand how an oracle network is
+configured and assess whether it can be trusted. Generally, oracles should only
+be used for non-mission-critical functions, and worst-case scenarios should
+always be accounted for.
 
 ### Oracles on Solana
 
-There are many
-[Oracle providers on Solana](https://solana.com/ecosystem/explore?categories=oracle).
-Two of the most well known are [Pyth](https://pyth.network) and
-[Switchboard](https://switchboard.xyz). They’re each unique and follow slightly
-different design choices.
-
-**Pyth** is primarily focused on financial data published from top-tier
-financial institutions. Pyth’s data providers publish the market data updates.
-These updates are then aggregated and published onchain by the Pyth program. The
-data sourced from Pyth is not completely decentralized as only approved data
-providers can publish data. The selling point of Pyth is that its data is vetted
-directly by the platform and sourced from financial institutions, ensuring
-higher quality.
-
-**Switchboard** is a completely decentralized oracle network and has data of all
-kinds available. Check out all of the feeds
-[on their website](https://app.switchboard.xyz/solana/devnet/explore)
-Additionally, anyone can run a Switchboard oracle and anyone can consume their
-data. This means you'll have to be diligent about researching feeds. We'll talk
-more about what to look for later in the lesson.
-
-Switchboard follows a variation of the stake weighted oracle network described
-in the third option of the previous section. It does so by introducing what are
-called TEEs (Trusted Execution Environments). TEEs are secure environments
-isolated from the rest of the system where sensitive code can be executed. In
-simple terms, given a program and an input, TEEs can execute and generate an
-output along with a proof. If you’d like to learn more about TEEs, please read
-[Switchboard’s documentation](https://docs.switchboard.xyz/functions).
-
-By introducing TEEs on top of stake weighted oracles, Switchboard is able to
-verify each oracle’s software to allow participation in the network. If an
-oracle operator acts maliciously and attempts to change the operation of the
-approved code, a data quote verification will fail. This allows Switchboard
-oracles to operate beyond quantitative value reporting, such as functions --
-running offchain custom and confidential computations.
+Solana has a rich ecosystem of oracle providers. Some notable ones include
+[Pyth Network](https://pyth.network), [Switchboard](https://switchboard.xyz),
+[Chainlink](https://chain.link), and
+[DIA](https://www.diadata.org/solana-price-oracles/). Two of the most well-known
+are [Pyth](https://pyth.network) and [Switchboard](https://switchboard.xyz),
+each with unique design choices.
```

Comment:
There's some duplication here from the old text. I suggest:

```
Solana has a rich ecosystem of oracle providers. Some notable ones include:

 - [**Pyth**](https://www.pyth.network/price-feeds) ...
 - [**Switchboard**](https://example.com) ...
 - [**Chainlink**](https://example.com) ...
 - [DIA](https://www.diadata.org/solana-price-oracles/) ...
 
 We'll use Switchboard in this lesson, but the concepts are common to most oracles and you should pick the oracle that best meets your needs. 
```

Reply:
Yes resolved as per suggestion.

Reply:
Sorry @0xCipherCoder I should be more specific.

Don't do:
```
some text

 - Pyth
 - Switchboard
 - Chainlink
 - Dia 

some text

 - Pyth
 - Switchboard
```
Instead, replace all of that with:

```
some text

 - Pyth
 - Switchboard
 - Chainlink
 - Dia 
```

I hope that makes sense.

Reply:
Just curious should we then remove description that has been given again for Pyth and Switchboard? Just keep the original oneliner description then go ahead with switchboard.

Reply:
Yes - keep all the information but use a single list.

Reply:
Updated content list as per suggestion.